### ISSUE TITLE
Network inaccessible error when using a WiFi network

### ISSUE THREAD

**Issue by raananb** [sqlitebrowser/sqlitebrowser#1595]
Details for the issue
--------------------
Upon startup, an error message is displayed: The network is not accessible. DB Browser then works correctly.

#### What did you do?
Installed DB Browser for SQLite-win64.msi 
The same problem arose with yesterday's (and maybe the day before).
#### What did you see instead?

Useful extra information
-------------------------
> DB4S v3.11.99 (Oct 28 2018) on Windows 10 (10.0) (winnt/10.0.17134) [x86_64]
> using SQLite Version 3.25.2
> and Qt 5.10.1

**Comment by justinclift** [comment]
Thanks @raananb.  Yep, we're in the process of upgrading the version of Qt we use, and it looks like we need to use a newer version of MSVC to make that go away.

We're figuring that out now, but it'll probably take a few more days - maybe even until the weekend (best guess).

As a workaround, if you turn off the "Check for updated version" (or similar wording) in the application preferences, the error message shouldn't happen.

Does that help? :smile:

**Comment by raananb** [comment]
With "Auto update" unchecked, the error message is not displayed on today's nightly (3.11.99 Oct 29 2018).

**Comment by chrisjlocke** [comment]
Closing, as this is resolved (technically....) and has a workaround... (and is also mentioned elsewhere)

**Comment by RudolfJan** [comment]
I have the same issue. The workaround helps.

**Comment by chrisjlocke** [comment]
The latest version of the nightlies is also useful ... depending on your operating system, of course.

**Comment by mgrojo** [comment]
And the fix will enter our next beta in the v3.11.x branch.

77a701dc979ecf6213bd14c98e87006f8cb82cb5

**Comment by vlakoff** [comment]
Hi, I just tried the recently released 3.11.0 and encountered this issue.

What's interesting is that I encounter it with `DB.Browser.for.SQLite-3.11.0-win64.zip` but not with `DB.Browser.for.SQLite-3.11.0-win32.zip`.

**Comment by karim** [comment]
@vlakoff I just downloaded both versions now and I can't reproduce this error.

This error means you are missing a Qt DLL (**qgenericbearer.dll**) file. Can you check if you have **bearer** folder and this file inside it?

The error is not happening in 32-bit version because we use different version of Qt for 32-bit (5.7.1) and 64-bit (5.11.3). Qt 5.7.1 doesn't come with this file and it is not needed.

**Comment by justinclift** [comment]
Uh oh.  Hopefully the packaging for Win64 isn't busted.  If it is, we'll be making a v2 package today... :wink:

**Comment by vlakoff** [comment]
Indeed I have dug a bit about this issue, and understood there are different Qt versions. Hence only 64-bit is affected.

As you can see in the released zip, yes there is a `bearer\qgenericbearer.dll` file. But maybe it isn't loaded? Just in case, I tried putting copies of the DLL everywhere in the folder structure, but it didn't fix the issue.

I also tried clearing the program settings from the Registry, and also installing using the MSI, but issue still present.

And obvisouly, I'm using a wifi connection.

**Comment by karim** [comment]
@vlakoff Thanks for clarifications.

> As you can see in the released zip, yes there is a `bearer\qgenericbearer.dll` file.

My initial thought, as well as Justin's I think, is that the .zip file was damaged somehow so it wasn't extracting the file or it is missing. Clearly this isn't the case here because the MSI is showing the same error also.

> I also tried clearing the program settings from the Registry, and also installing using the MSI, but issue is still present.

Turning off the automatic update check at startup should make the error go away for now.

> And obviously, I'm using a wifi connection.

While we are trying to narrow it down, can you try connecting to a wired network and see if the error is gone.

@justinclift Would it be possible to make a one-time 64-bit build *without* this commit https://github.com/sqlitebrowser/sqlitebrowser/commit/73946400c32d1f7cfcd4672ab0ab3f563eb84f4e?

I suspect that this workaround, while it fixed the high ping issue, it disabled networking polling (check & reconnect) on wifi. I could be wrong, but I'm out of ideas here. :smiley:

**Comment by vlakoff** [comment]
I'm lazy to move my whole desktop computer and the internet router because all I have is a very short ethernet cable ^^ However, I tried using the internet connection of my phone (USB tethering), disabling the wifi adapter, and the issue doesn't happen.

Interestingly enough, I don't have the issue on another computer, which is connected using wifi.

I guess the commited fix/workaround just doesn't cover 100% of cases. I'm hoping for the proper upstream fix to land, on the Qt side.

**Comment by vlakoff** [comment]
How about this suggestion I read from the Qt ticket?
> Just check whether you have a connection with the server without checking the network state of the device itself. Checking the state of the connection with the server is a much better indication of whether your app is online or not.

**Comment by justinclift** [comment]
> Would it be possible to make a one-time 64-bit build without this commit 73946400c32d1f7cfcd4672ab0ab3f563eb84f4e?

Yeah, that's do-able.  Still need it done?  And if so, 32-bit or 64-bit? :smile:

**Comment by karim** [comment]
>  Still need it done?

Yes, please. Like it was mentioned by @vlakoff the workaround seems to be inconsistent and doesn't cover all the cases. Sometimes it works, sometimes not. I guess it is caused by different wifi configurations.

If that fixed the connection issue then I think our fix should be in the wiki and not in the code.

This answer https://apple.stackexchange.com/a/312388 points out that we can actually set it as an environment variable to fix the Qt problem for all Qt applications. Or if you want it for a specific application, then something like...

```
set QT_BEARER_POLL_TIMEOUT=-1
DB4S.exe
``` 

> And if so, 32-bit or 64-bit? :smile:

64-bit since this isn't a problem with Qt 5.7.1 that is being used with the 32-bit version.

**Comment by vlakoff** [comment]
It would be interesting to know if other people than me encounter the issue

**Comment by justinclift** [comment]
No worries, I'll get this done today.  The box which automatically generates our nightly builds, will start generating them in roughly 1/2 and hour.  That will tie it up for roughly 3 hours, after which I'll be able to use it for making one-off builds like this. :smile:

**Comment by justinclift** [comment]
Win64 build:

&nbsp; &nbsp; https://nightlies.sqlitebrowser.org/win64-onceoffs/DB.Browser.for.SQLite-issue595v1-win64.msi

There's a .zip version too, if that's better for you than an .msi:

&nbsp; &nbsp; https://nightlies.sqlitebrowser.org/win64-onceoffs/DB.Browser.for.SQLite-issue595v1-win64.zip

**Comment by karim** [comment]
Thanks @justinclift :smiley: 

@vlakoff Would you mind giving it a try now and see if this version is better?

**Comment by torpkev** [comment]
Hi - I still get the Error: The network is not accessible witht he Win64 build that @justinclift posted on Windows 10

Unticking Automatic Updates still works as a workaround.

**Comment by Krivvenz** [comment]
I have just upgraded to 3.11 and can confirm that I am also getting the "The network is not accessible" message on opening. 64 bit version and turning off automatic updates silences the error.

**Comment by karim** [comment]
Thanks @torpkev and @Krivvenz for reporting this. I reopened the issue now.

Just to confirm, You are on Windows 10, using the 64-bit version, and on wifi connection, right?

**Comment by Krivvenz** [comment]
Hi @karim - I can confirm all that information is correct - Windows 10, 64 bit and a WiFi connection.

**Comment by karim** [comment]
Thanks @Krivvenz , I added this to our list of known issues here #1688.

**Comment by vlakoff** [comment]
I have tried the [custom zip](https://github.com/sqlitebrowser/sqlitebrowser/issues/1595#issuecomment-462123714), I also encounter the issue with it. BTW I'm on Windows 7.

**Comment by justinclift** [comment]
@vlakoff @torpkev @Krivvenz If you open a web browser and try to load this (super simple) text file, does it work:

&nbsp; &nbsp; https://download.sqlitebrowser.org/currentrelease

That's the file which DB4S is trying to read when it starts up, and failing on.

Is there any chance it's being blocked by some firewall setting?  Another possibility is perhaps needing to use a proxy to access the internet (common in corporate environments).  We haven't yet added proxy support, though it's on our ToDo list for "sometime in the future". :wink:

**Comment by vlakoff** [comment]
I have no trouble at all to reach this file.

As I commented above, can't you do a simple request to the server rather than looking at the network adapters? That seems pretty unusual for an application.

**Comment by justinclift** [comment]
> can't you do a simple request to the server

I think that's what we used to do, but it turned out that (by default) Qt itself does some weird stuff, as per #1209.

So, the patch @karim points at above was a workaround to that weird stuff.

Not sure what the right approach here will be, but @MKleusberg will hopefully have some ideas. :smile:

**Comment by chrisjlocke** [comment]
Just a nod that I'm getting this on my Win 10 laptop at home, as I've recently switched from LAN to WiFi.
Can confirm no proxy stuff in the way, and can open the previous link directly.  
Also get the error on the custom ZIP posted.

**Comment by Krivvenz** [comment]
@justinclift I can read the file fine too: 

3.11.0
https://sqlitebrowser.org/blog/version-3-11-0-released

I'm on my personal home laptop, so no proxy or firewall issues :).

**Comment by justinclift** [comment]
Well, I'm mostly at a loss then.  This is all on Windows (7, 10, etc) yep?

eg not being reported to happen (so far) on macOS, Linux, others?

If that's the case, then maybe it really is as simple as some kind of packaging problem, and there's an additional .dll or similar in Qt 5.11.3 that needs including.

@chrisjlocke When you have a few moments, maybe try installing Qt 5.11.3 to your Win 10 laptop at home, then 100% of it's deployment .dlls and put them in the DB4S install directory alongside the other ones.  If the error goes away, we've found a smoking gun. :wink:

**Comment by justinclift** [comment]
Not sure if it's related, but when doing some testing of Win64 builds on a local Win7 development VM, I noticed it complaining (various SSL cipher errors) upon startup, and it also couldn't load the remote file.  Looked like a library problem of some sort.  Note that it wasn't with a proper install of DB4S though, so it could be something different.

But, it's something to keep in mind for investigating.

**Comment by karim** [comment]
@justinclift 32-bit seems to be unaffected. Do you think building a 64-bit version with Qt 5.7.1 (same as 32-bit version) will make any difference?

**Comment by justinclift** [comment]
Interesting idea.  Yeah, that'd be a good approach.  If it works, it'd be a useful data point for figuring out what's wrong.

It'll probably need to be done with the 64-bit part of MSVC 2013, as Qt 5.7.1 didn't have support for MSVC 2017.  Probably need to install the 64-bit Qt 5.7.1 support again, but that's not difficult.

I'll try to get this done some time in the next few hours. :smile:

**Comment by justinclift** [comment]
Having real trouble with this.  Running the Qt installer, it keeps on picking some mirror in China that doesn't have the Qt 5.7.1 bits and it point blank **refuses** to allow selection of any other mirror.

eg I can't download the files.

Will come back to this tomorrow, there's probably some simple solution I'm just not thinking of.  Mentally tired atm. :frowning:

**Comment by karim** [comment]
Use the offline installer.

- https://download.qt.io/archive/qt/5.7/5.7.1/

**Comment by justinclift** [comment]
Apparently the offline installers don't play well with an existing installation.

That being said, I haven't actually tried it... could be completely fine.

Meh, the VM can be snapshotted before hand anyway, so won't hurt to try it. :smile:

**Comment by justinclift** [comment]
Getting the build process set up now. :wink:

@karim Noticed something interesting, and potentially relevant, though not sure to what extent yet.

Looking at this:

https://github.com/sqlitebrowser/sqlitebrowser/blob/f5f5894e138300d99d37702ba05b19d973b6149c/installer/windows/product.wxs#L65-L70

It seems to exclude the bearer plugin from the 32-bit build (which seems ok).  But the comment makes it sound like that dll doesn't exist with Qt 5.7.1.  Looking at the freshly installed Qt 5.7.1 (msvc2013_64 dir), it definitely does.

I'll exclude it from this build for this attempt :wink:, so it more closely matches the behaviour of the 32-bit one.  But it potentially gives us another option to play around with later, to see if it makes a difference.

**Comment by justinclift** [comment]
Well, it built and seems to run ok on the VM it was built on:

* https://nightlies.sqlitebrowser.org/win64-onceoffs/DB.Browser.for.SQLite-issue1595v2-win64.msi
* https://nightlies.sqlitebrowser.org/win64-onceoffs/DB.Browser.for.SQLite-issue1595v2-win64.zip

These builds may have the wrong version of the merge module in them (not sure), so if it fails to launch... that'd be why.

**Comment by chrisjlocke** [comment]
v3 surely? 😉 
Again, they run, open databases created earlier, and encrypt/open new databases.

**Comment by karim** [comment]
> But the comment makes it sound like that dll doesn't exist with Qt 5.7.1. Looking at the freshly installed Qt 5.7.1 (msvc2013_64 dir), it definitely does.

At first I did include it in both builds. Then later found out that the 32-bit application starts normal without it, so I removed it. I attributed this to the fact that this plugin isn't needed for the older Qt version that we use for the 32-bit version. Sorry if my comments were confusing. :smiley:

**Comment by justinclift** [comment]
No worries.  It gives us another potential knob to turn, to try and fix things. :smile:

**Comment by vlakoff** [comment]
I don't encounter the issue with the `issue1595v2-win64.zip` :-)

**Comment by justinclift** [comment]
Thanks @vlakoff.  If I make a v3 sometime tomorrow with some changes - so we can try and narrow down what the problem is - would you be ok to try it out + report back?

**Comment by vlakoff** [comment]
Of course, it's quite fast to test.

**Comment by justinclift** [comment]
Excellent.  Thanks @vlakoff. :smile:

**Comment by karim** [comment]
> I don't encounter the issue with the `issue1595v2-win64.zip` :-)

..and we have a winner ! :smile: :tada: 

Without any change of code it's more clear that this is a Qt problem. Now we need to figure out which version of Qt is causing this.

I guess we have to rebuild the application with different versions of Qt to find out, and I think this is what Justin will be doing. :smiley:

**Comment by justinclift** [comment]
Yeah, next one to test will be the same Qt 5.7.1, but this time including that qgenericbearer.dll, as it sounds close to the problem. :wink:

**Comment by justinclift** [comment]
Win64 v3 build:

* https://nightlies.sqlitebrowser.org/win64-onceoffs/DB.Browser.for.SQLite-issue1595v3-win64.msi
* https://nightlies.sqlitebrowser.org/win64-onceoffs/DB.Browser.for.SQLite-issue1595v3-win64.zip

The only difference between this and the v2 build, is this one has the qgenericbearer dll in it.  Lets see if the presence of that triggers the network issue. :smile:

Also, for anyone using the .msi version, it's probably best to manually uninstall the existing DB4S installation.  eg Control Panel -> Add/Remove programs in Win7 (not sure where in Win10, as I've never used it)

**Comment by vlakoff** [comment]
* `issue1595v1-win64.zip`: encounter issue
* `issue1595v2-win64.zip`: doesn't encounter issue
* `issue1595v3-win64.zip`: doesn't encounter issue

Would it be an option to stick on the older Qt for now, as a "lesser evil"?

**Comment by justinclift** [comment]
Thanks @vlakoff, that's super useful. :smile:

> Would it be an option to stick on the older Qt for now, as a "lesser evil"?

It's definitely an option.  But, let's try things with a few different versions of Qt first, to see if we can isolate where the problem starts and figure out what's really causing it.

I'll make some builds using different Qt versions over the next few hours. :smile:

**Comment by karim** [comment]
> But, let's try things with a few different versions of Qt first, to see if we can isolate where the problem starts

Betting time now people! I'd say the problem started with Qt 5.10. :smile:

**Comment by justinclift** [comment]
Currently building with Qt 5.9.7, as that's right in the middle:

* [x] Qt 5.7 - working
* [ ] Qt 5.8
* [ ] Qt 5.9
* [ ] Qt 5.10
* [x] Qt 5.11 - network error

**Comment by justinclift** [comment]
Qt 5.9.7 build:

* https://nightlies.sqlitebrowser.org/win64-onceoffs/DB.Browser.for.SQLite-issue1595v4-Qt5.9.7-win64.msi
* https://nightlies.sqlitebrowser.org/win64-onceoffs/DB.Browser.for.SQLite-issue1595v4-Qt5.9.7-win64.zip

**Comment by vlakoff** [comment]
Encountering the issue :)

**Comment by justinclift** [comment]
Thanks @vlakoff.  k, will do a Qt 5.8 build next. :smile:

**Comment by justinclift** [comment]
Interesting side note.  Both of the builds with the error were created using MSVC 2017 x64.  Wonder if that's somehow the cause... the Qt 5.8 build (working on it how) will use the old MSVC 2013, same at the Qt 5.7.1 build.

**Comment by justinclift** [comment]
Qt 5.8 build:

* https://nightlies.sqlitebrowser.org/win64-onceoffs/DB.Browser.for.SQLite-issue1595v5-Qt5.8.0-win64.msi
* https://nightlies.sqlitebrowser.org/win64-onceoffs/DB.Browser.for.SQLite-issue1595v5-Qt5.8.0-win64.zip

**Comment by vlakoff** [comment]
5.8 working :)

**Comment by justinclift** [comment]
Thanks @vlakoff.  Looks like whatever the problem is, it's either something to do with the change to Qt 5.9 or something to do with the change to MSVC 2017.

**Comment by karim** [comment]
> Looks like whatever the problem is, it's either something to do with the change to Qt 5.9 or something to do with the change to MSVC 2017.

That narrowed it down a lot. It is a bit hard to go from here since Qt 5.9 dropped support for MSVC 2013 so we can't use that, and Qt 5.8 does not support MSVC 2017.

Although I heard we can use MSVC 2017 with Qt 5.8 msvc2015 files because [MSVC 2015 and MSVC 2017 are somehow compatible](https://docs.microsoft.com/en-us/cpp/porting/binary-compat-2015-2017?view=vs-2017).

Also, does building with Qt 5.9.0 will be the same as 5.9.7? Asking because it might be helpful to know the exact version this bug was introduced.

**Comment by justinclift** [comment]
Downloading the MSVC 2015 DVD presently.  Going to experiment a bit. :wink:

**Comment by justinclift** [comment]
Noticed something interesting.  Although Qt 5.9.7 has no support for MSVC 2013.... Qt 5.9.6 does.

I'll make a Win64 build using Qt 5.9.6 + MSVC 2013, and lets see if it works.  If it does, that points to the MSVC 2017 change being the problem.

**Comment by justinclift** [comment]
Oh... Qt 5.10 has MSVC 2013 support still too.

**Comment by justinclift** [comment]
... and Qt 5.9.7 does have MSVC 2013 support.  It's just capitalised differently than expected, so shown out of order in the list. :wink:

**Comment by justinclift** [comment]
k, new build to try:

* https://nightlies.sqlitebrowser.org/win64-onceoffs/DB.Browser.for.SQLite-issue1595v6-Qt5.9.7-MSVC2013-win64.msi
* https://nightlies.sqlitebrowser.org/win64-onceoffs/DB.Browser.for.SQLite-issue1595v6-Qt5.9.7-MSVC2013-win64.zip

If this one works, it means the error is something to do with the MSVC2017 change.

**Comment by vlakoff** [comment]
I encounter the issue with it

**Comment by justinclift** [comment]
Excellent @vlakoff. :smile:

That means the problem really is something to do with the Qt version change, and isn't to do with the change in compiler.

This could be tricky. :frowning:

We might need to go with your suggestion yesterday - make the 3.11.1 release using a Qt version that works - and figure out the proper solution separately, in time for the next release.

**Comment by justinclift** [comment]
@vlakoff As an experiment, with that "v6" version above, if you delete the "qgenericbearer.dll" file from it after installation... does the error still happen?

**Comment by vlakoff** [comment]
There is still the error if I delete the file.

**Comment by justinclift** [comment]
Thanks @vlakoff.  Looks like it's not something to do with that .dll then.

This is fairly frustrating. :frowning:

**Comment by vlakoff** [comment]
Just to make sure, are you sure the dll is really loaded?

**Comment by justinclift** [comment]
Nope, no longer sure at all.

We added it a while ago, as it seemed to solve a similar network error + the Qt docs say it's needed.

But it seems like it's not doing anything useful at the moment.

Hmm... unless there's something wrong with how we're deploying it with Qt 5.9 and later maybe (not sure, will check).  I'm trying to fix a macOS problem at the moment, so will circle back to this again after that.

**Comment by karim** [comment]
@vlakoff 

> Just to make sure, are you sure the dll is really loaded?

Yes, I think it is loaded. All the posted versions here work for me (since I'm not on a wifi connection) and if I remove this dll then I get an error.

Earlier you pointed out...

> Interestingly enough, I don't have the issue on another computer, which is connected using wifi.

Is it possible that this could be related to the network/wifi configuration?

After reading the answer to this [question](https://forum.qt.io/topic/91204/some-trouble-with-qnetworkaccessmanager-and-wifi-on-windows-7), it pointed out that maybe you have multiple network interfaces and Qt is checking the wrong one (Ethernet while the cable is unplugged) so it returns "not accessible".

This might explain why it worked on one computer and not the other. Maybe Qt got the right interface on one but not the other.

If it is not too much to ask, can you try disabling other unused network interfaces and see if it starts working? Also, can you check the networking info for both computers (the one that worked and the one that didn't) and try to find if there is anything *interesting* there. :smiley: 

@justinclift This is likely a Qt bug. Can you try building another version using Qt 5.9.0 (the next one right after Qt 5.8.0) just to know if it is something introduced with 5.9 or one of the patches in 5.9.x.

**Comment by justinclift** [comment]
Sure, can do.  The build box is currently doing some macOS building though, so it'll need to come after that. :smile:

**Comment by karim** [comment]
More details...

This bug (https://bugreports.qt.io/browse/QTBUG-68613) is like the one we have here. It appears there was a plugin (qnativewifibearer.dll) in earlier versions of Qt but it was removed (https://bugreports.qt.io/browse/QTBUG-40332) because it was causing high ping, in favor of improving it (https://bugreports.qt.io/browse/QTBUG-65586). None of these issues are resolved.

The last comment here (https://bugreports.qt.io/browse/QTBUG-66461) points out that *if* we found this dll (qnativewifibearer.dll) and put it in the bearer folder, then all our problems will be solved. :smiley: 

It is very hacky, and I don't even know which version of Qt has this file, but I write it here as a data point.

**Comment by justinclift** [comment]
Interesting.  It sounds like Qt 5.10.0 will have the dll.  I'll have a play around with that later. :smile:

**Comment by justinclift** [comment]
This bizarre network level stuff from Qt is really... unsatisfactory.  I'd heard of other places implementing their own (basic) network functionality, so they don't have to use any of Qt's networking.  I'm starting to understand why. Heh Heh Heh.

**Comment by vlakoff** [comment]
This definitely could be related to network configuration differences (notably the adapters) between the two computers. For instance, my computer n°2 (the unaffected one here) is the only one to have an issue in another program because of some network configuration.

My computer n°1 (the affected one here) has an ethernet adapter which was already disabled. Also some VPN adapters, that I tried disabling/uninstalling, without change, and I doubt they are the culprit otherwise computer n°2 would also have the issue.

I read the following in https://doc.qt.io/qt-5/qnetworkconfigurationmanager.html#allConfigurations:
> Therefore, if WLAN configurations are of interest, it is recommended that [updateConfigurations](https://doc.qt.io/qt-5/qnetworkconfigurationmanager.html#updateConfigurations)() is called once after [QNetworkConfigurationManager](https://doc.qt.io/qt-5/qnetworkconfigurationmanager.html) instantiation (WLAN scans are too time consuming to perform in constructor).

**Comment by justinclift** [comment]
@MKleusberg Does the above item about "updateConfigurations()" seem like it'd be useful to us?

It looks like this particular issue (1595) is some Qt specific weirdness, starting in the Qt 5.9 series that only happens with WiFi connected computers.  If we can get things to work with computers connected via WiFi - and really, the connection tech shouldn't make any difference - that'd be good. :smile:

**Comment by vlakoff** [comment]
I don't know Qt, but I thought of the following:

```
configurationManager = new QNetworkConfigurationManager;
configurationManager.updateConfiguration();

on signal updateCompleted {
    accessManager = new QNetworkAccessManager;
    accessManager.setConfiguration(configurationManager.defaultConfiguration());
    // Perform update check
}
```

(disclaimer: completely pseudo-code!)

**Comment by justinclift** [comment]
Yeah, we have existing network setup code.  No calls in it to `QNetworkConfigurationManager()` though.  First thing I checked, prior to pinging @MKleusberg above. :wink:

**Comment by MKleusberg** [comment]
Oh the joys of Qt networking. I don't yet understand how the pseudo code above would help. The default network configuration which ```QNetworkAccessManager``` uses is the one returned by ```QNetworkConfigurationManager:::defaultConfiguration()```. And I don't think the default configuration should change by calling ```QNetworkConfigurationManager::updateConfigurations()``` because scanning for WLAN connections doesn't change your personal configuration. Or am I missing something :smile:

**Comment by justinclift** [comment]
Sure, the theory is good... but (if it's not hard) maybe we could try doing an `updateConfigurations()` once anyway, just to see if it makes a difference?

And yeah I get that it shouldn't... but just in case. :wink:

**Comment by MKleusberg** [comment]
No worries :smile: Let's try - the Qt network stuff can definitely be unpredictable :wink: 

Here's the patch, can you make a build with that?
```diff
diff --git a/src/MainWindow.cpp b/src/MainWindow.cpp
index 848c1acd..71f15292 100644
--- a/src/MainWindow.cpp
+++ b/src/MainWindow.cpp
@@ -518,15 +518,6 @@ void MainWindow::init()
     // Load all settings
     reloadSettings();
 
-#ifdef CHECKNEWVERSION
-    // Check for a new version if automatic update check aren't disabled in the settings dialog
-    if(Settings::getValue("checkversion", "enabled").toBool())
-    {
-        m_remoteDb->fetch("https://download.sqlitebrowser.org/currentrelease",
-                          RemoteDatabase::RequestTypeNewVersionCheck);
-    }
-#endif
-
 #ifndef ENABLE_SQLCIPHER
     // Only show encryption menu action when SQLCipher support is enabled
     ui->actionEncryption->setVisible(false);
diff --git a/src/RemoteDatabase.cpp b/src/RemoteDatabase.cpp
index a67c989b..0b5dc867 100644
--- a/src/RemoteDatabase.cpp
+++ b/src/RemoteDatabase.cpp
@@ -1,5 +1,6 @@
 #include <QApplication>
 #include <QtNetwork/QNetworkAccessManager>
+#include <QtNetwork/QNetworkConfigurationManager>
 #include <QMessageBox>
 #include <QtNetwork/QNetworkReply>
 #include <QFile>
@@ -21,9 +22,22 @@
 
 RemoteDatabase::RemoteDatabase() :
     m_manager(new QNetworkAccessManager),
+    m_configurationManager(new QNetworkConfigurationManager),
     m_progress(nullptr),
     m_dbLocal(nullptr)
 {
+    // Update network configurations
+    connect(m_configurationManager, &QNetworkConfigurationManager::updateCompleted, [this]() {
+        m_manager->setConfiguration(m_configurationManager->defaultConfiguration());
+
+#ifdef CHECKNEWVERSION
+    // Check for a new version if automatic update check aren't disabled in the settings dialog
+    if(Settings::getValue("checkversion", "enabled").toBool())
+        fetch("https://download.sqlitebrowser.org/currentrelease", RemoteDatabase::RequestTypeNewVersionCheck);
+#endif
+    });
+    m_configurationManager->updateConfigurations();
+
     // Set up SSL configuration
     m_sslConfiguration = QSslConfiguration::defaultConfiguration();
     m_sslConfiguration.setPeerVerifyMode(QSslSocket::VerifyPeer);
diff --git a/src/RemoteDatabase.h b/src/RemoteDatabase.h
index bcba03e2..8a14a7a3 100644
--- a/src/RemoteDatabase.h
+++ b/src/RemoteDatabase.h
@@ -5,6 +5,7 @@
 #include <QtNetwork/QSslConfiguration>
 
 class QNetworkAccessManager;
+class QNetworkConfigurationManager;
 class QString;
 class QNetworkReply;
 class QSslError;
@@ -88,6 +89,7 @@ private:
     void clearAccessCache(const QString& clientCert);
 
     QNetworkAccessManager* m_manager;
+    QNetworkConfigurationManager* m_configurationManager;
     QProgressDialog* m_progress;
     QSslConfiguration m_sslConfiguration;
     QMap<QString, QSslCertificate> m_clientCertFiles;
```

**Comment by justinclift** [comment]
Will try in a bit.  The mac mini is tied up at the moment in the middle of a long (~hour) compilation job. :innocent:

**Comment by justinclift** [comment]
Chucked that patch into a commit (https://github.com/sqlitebrowser/sqlitebrowser/commit/0116181a23c127a2a1bf066ffbef391cedb07c40) in `issue1595v7` branch.  When the mac mini is finished it's current build stuff, I'll try a Qt 5.9.7 (known broken) build with this, to see if it becomes unbroken. :smile:

**Comment by vlakoff** [comment]
The updateConfigurations/updateCompleted being async, with this code wouldn't the version check be executed before the updateCompleted signal is received?

**Comment by vlakoff** [comment]
Nevermind, I read the diff too fast

**Comment by vlakoff** [comment]
Just in case, refs:
 https://stackoverflow.com/questions/51686943/no-qnetworkconfiguration-although-i-can-connect-to-wifi

(amongst many, many other reports/questions that can found on the web about this issue…)

**Comment by vlakoff** [comment]
(see workaround in the answer)

**Comment by justinclift** [comment]
Oh, looks like an interesting approach.  Guess we could try it.

In the meantime, a build with Martin's patch (above) is currently in progress. :smile:

**Comment by karim** [comment]
The answer in the link suggests using something like..

`networkAccessManager->setNetworkAccessible(QNetworkAccessManager::Accessible);`

.. but Qt [documentation](https://doc.qt.io/qt-5/qnetworkaccessmanager.html#networkAccessible-prop) about this function says..

> Note: Calling setNetworkAccessible() does not change the network state.

This function is supposed to be used like this..

`networkAccessManager->setNetworkAccessible(QNetworkAccessManager::NotAccessible);`

.. to disable all network requests by setting the access manager to not accessible (so all network requests will fail), and then use it again later to enable it, but it does not actually make it accessible if it is not.

**Comment by justinclift** [comment]
Hmmm, in the [follow up question] link to that post, it mentions this:

> I understand I can compile a Qt version without bearer management,

That could be a useful direction to experiment too.  :smile:

**Comment by justinclift** [comment]
New build, this one with Martin's patch from above:

* https://nightlies.sqlitebrowser.org/win64-onceoffs/DB.Browser.for.SQLite-issue1595v7-Qt5.9.7-MSVC2013-win64.msi
* https://nightlies.sqlitebrowser.org/win64-onceoffs/DB.Browser.for.SQLite-issue1595v7-Qt5.9.7-MSVC2013-win64.zip

Really hoping this one works decently. :smile:

**Comment by karim** [comment]
> I understand I can compile a Qt version without bearer management,

At some point I think we should consider compiling Qt ourselves if it's not too hard. It might help with few issues, like using the latest OpenSSL for example.

**Comment by justinclift** [comment]
Yeah, am thinking the same.

**Comment by vlakoff** [comment]
v7 works :)

**Comment by justinclift** [comment]
Well, now that's something. :smile:

**Comment by justinclift** [comment]
k, lets try a build of Qt 5.11.3 - the one I'd really like to use for Win64 :wink: - with that patch, and see if it still works.  If it does, then we probably have a winner for now.

Well, as long as it still works ok for everyone else's network connection (eg ethernet) too. :smile:

v8 coming up in a bit...

**Comment by justinclift** [comment]
v8 build:

* https://nightlies.sqlitebrowser.org/win64-onceoffs/DB.Browser.for.SQLite-issue1595v8-Qt5.11.3-MSVC2017-win64.msi
* https://nightlies.sqlitebrowser.org/win64-onceoffs/DB.Browser.for.SQLite-issue1595v8-Qt5.11.3-MSVC2017-win64.zip

This has Martin's patch too, but is based on Qt 5.11.3, and is compiled with MSVC 2017.

**Comment by drescherjm** [comment]
I just tested the v8 build and it worked for me without the error message on Win10 1803.

**Comment by justinclift** [comment]
Excellent @drescherjm!  That's really good news.

Next step, lets try it with the v3.11.x branch.  I'll make a build with that soon, if the patch applies cleanly there.  Just waiting for some other stuff to finish compiling. :smile:

**Comment by vlakoff** [comment]
v8 works fine here too :)

**Comment by justinclift** [comment]
v9 build.  This is Martins' patch on our `3.11.x` code, built using Qt 5.11.3 + MSVC 2013.

eg if this is confirmed working on both WiFi and ethernet... we can probably regard this as solved. :smile:

**Comment by karim** [comment]
It's working fine here on Ethernet. It looks like **qgenericbearer.dll** (added in this commit https://github.com/sqlitebrowser/sqlitebrowser/commit/77a701dc979ecf6213bd14c98e87006f8cb82cb5) is no longer needed with this patch. Deleting the file has no effect on the application and it starts without a problem.

**Comment by vlakoff** [comment]
aaaand, v9 works here too :)

**Comment by MKleusberg** [comment]
Awesome :smile: 

Let's make one final test. @justinclift Can you build any of these binaries again but apply this patch on top of the other one?
```diff
diff --git a/src/RemoteDatabase.cpp b/src/RemoteDatabase.cpp
index 0b5dc867..7f13f896 100644
--- a/src/RemoteDatabase.cpp
+++ b/src/RemoteDatabase.cpp
@@ -30,6 +30,8 @@ RemoteDatabase::RemoteDatabase() :
     connect(m_configurationManager, &QNetworkConfigurationManager::updateCompleted, [this]() {
         m_manager->setConfiguration(m_configurationManager->defaultConfiguration());
 
+        QMessageBox::warning(nullptr, "test", "Working!");
+
 #ifdef CHECKNEWVERSION
     // Check for a new version if automatic update check aren't disabled in the settings dialog
     if(Settings::getValue("checkversion", "enabled").toBool())
```

By adding a message box I want to make sure that the ```updateCompleted``` signal is emitted at all. If the message box shows up in this build, we definitely know that the fix is working. If it doesn't show up, the code for the automatic update check is never actually called which would mean that the problem is still there, it's only better hidden. Let's hope it works :wink:

**Comment by justinclift** [comment]
No worries, v10 coming up shortly. :smile:

**Comment by justinclift** [comment]
v10 build is ready for testing:

* https://nightlies.sqlitebrowser.org/win64-onceoffs/DB.Browser.for.SQLite-issue1595v10-Qt5.11.3-MSVC2017-win64.msi
* https://nightlies.sqlitebrowser.org/win64-onceoffs/DB.Browser.for.SQLite-issue1595v10-Qt5.11.3-MSVC2017-win64.zip

**Comment by chrisjlocke** [comment]
![image](https://user-images.githubusercontent.com/7872218/52902284-9d31f400-3206-11e9-8693-449e9e5e036c.png)

**Comment by vlakoff** [comment]
Same here, I'm getting two popups. You'll probably want to fix this :)

About the application startup time, I can't say. Personally I don't feel it changed.
As a reminder, the added code is asynchronous. But maybe it's the debug popup that causes the delay.
Do you also have the delay with v8 or v9?

**Comment by chrisjlocke** [comment]
v8 is about a second.  Just seemed longer with this particular build.  I'm sure its nothing, and as its just a testing build, I'm not panicking too much just yet.... ;)

**Comment by justinclift** [comment]
The v9 build should be the "best" so far.  This v10 build was just a confirmation check (the pop up) to see if the added function really does get called.

The pop up does occur, so looks like that confirmation is good and Martins patch is a real fix. :smile:

**Comment by vlakoff** [comment]
Now you want to avoid checking for update twice at program startup ;)

And maybe it uncovered another problem: I don't think this RemoteDatabase constructor should be executed several times. I guess it's a bit costly performance-wise, with the SSL initializations and so on.

**Comment by justinclift** [comment]
Interesting point.  From a "lets get it perfect" perspective, you could be on to something. :smile:

From a "lets get 3.11.1 out the door" perspective, I'm thinking this is good enough currently, or pretty close to it.

@MKleusberg Looks like this patch (minus pop up) is working.  Should we commit it to `master` + `3.11.1` as is, or do you want to iterate on it a bit more?

**Comment by justinclift** [comment]
Hmmm, wonder if that "two pop ups" thing is something along the lines of it popping up once per network interface on the system?

eg for someone with just a single network interface (from the Qt perspective) it's only once.  Likewise for people with more, it'd also happen further times.

**Comment by justinclift** [comment]
> It looks like qgenericbearer.dll (added in this commit 77a701d) is no longer needed with this patch.

@karim I'd rather not risk it at this point.  We can experiment with the nightlies, but for the 3.11.1 release let's just leave it in.  Unlikely to hit further edge cases - where it does turn out to be needed - that way. :smile:

**Comment by vlakoff** [comment]
Whether I enable/disable my Ethernet adapter and my VPN virtual adapter, I have 2 popups.

**Comment by justinclift** [comment]
Thanks @vlakoff, that helps. :smile:

**Comment by MKleusberg** [comment]
I have just committed a slightly modified version of the patch (without the popup of course) to both the master and the v3.11.x branch. Thank you everybody for your patience here :smile: And special thanks to @vlakoff who posted the pseudo code which in fact turned out to fix the problem here  :+1: 


> And maybe it uncovered another problem: I don't think this RemoteDatabase constructor should be executed several times. I guess it's a bit costly performance-wise, with the SSL initializations and so on.

The constructor is only called once. I have just checked that again :smile: The problem was the extra ```QNetworkConfigurationManager::updateConfigurations()``` call. Apprently this function is also called in the constructor of ```QNetworkConfigurationManager```, so no manual extra call is needed.

**Comment by justinclift** [comment]
Thanks @MKleusberg! :grinning: 

New v11 build coming up shortly (using the `v3.11.x` branch).

**Comment by justinclift** [comment]
v11 build, hopefully the last one we need to do:

* https://nightlies.sqlitebrowser.org/win64-onceoffs/DB.Browser.for.SQLite-issue1595v11-win64.msi
* https://nightlies.sqlitebrowser.org/win64-onceoffs/DB.Browser.for.SQLite-issue1595v11-win64.zip

**Comment by vlakoff** [comment]
v11 working, as expected :)

@MKleusberg Understood, the key wasn't exactly to call `updateConfigurations()` (was already called actually) but to listen to the `updateCompleted` signal. Good finding!

Could it be possible to put back the update check code into `MainWindow::init` rather than digging it inside the `RemoteDatabase` constructor?

**Comment by justinclift** [comment]
> v11 working, as expected :)

Fantastic!

> Could it be possible to put back the update check code into MainWindow::init rather than digging it inside the RemoteDatabase constructor?

Can make a v12 without stress.  It's a well oiled process now. :wink:

**Comment by justinclift** [comment]
Meh, I'm calling this "done".  It's good enough for our purposes for 3.11.1.

Moving the call around is fine, but lets do that in the `master` branch only. :smile:

**Comment by MKleusberg** [comment]
Make sense, @raananb - just in case we do need two instances of the ```RemoteDatabase``` class at some point in the future. I have just added a commit which moved the check back into the main window :smile:

**Comment by justinclift** [comment]
@MKleusberg Should that be "vlakoff"?

**Comment by MKleusberg** [comment]
@justinclift You're right - sorry about that :smile: Stupid Github auto completion...
